libs/openFrameworks/sound/ofRtAudioSoundStream.cpp

```diff
@@ -91,39 +91,58 @@ void ofRtAudioSoundStream::setOutput(ofBaseSoundOutput * soundOutput){
 
 //------------------------------------------------------------------------------
 bool ofRtAudioSoundStream::setup(int outChannels, int inChannels, int _sampleRate, int _bufferSize, int nBuffers){
-	if( audio != nullptr ){
+	return setup(outChannels, inChannels, _sampleRate, _bufferSize, nBuffers, OF_RT_API_UNSPECIFIED);
+}
+
+//------------------------------------------------------------------------------
+bool ofRtAudioSoundStream::setup(ofBaseApp * app, int outChannels, int inChannels, int sampleRate, int bufferSize, int nBuffers){
+	setInput(app);
+	setOutput(app);
+	return setup(outChannels,inChannels,sampleRate,bufferSize,nBuffers, OF_RT_API_UNSPECIFIED);
+}
+
+bool ofRtAudioSoundStream::setup(int outChannels, int inChannels, int _sampleRate, int _bufferSize, int nBuffers, ofRtAPI api)
+{
+	if (audio != nullptr) {
 		close();
 	}
 
-	nInputChannels		= inChannels;
-	nOutputChannels		= outChannels;
+	nInputChannels = inChannels;
+	nOutputChannels = outChannels;
 
-	sampleRate			=  _sampleRate;
-	tickCount			=  0;
-	bufferSize			= ofNextPow2(_bufferSize);	// must be pow2
+	sampleRate = _sampleRate;
+	tickCount = 0;
+	bufferSize = ofNextPow2(_bufferSize);	// must be pow2
 
 	try {
-		audio = shared_ptr<RtAudio>(new RtAudio());
-	} catch (std::exception &error) {
-   		ofLogError() << error.what();
+		if (api == OF_RT_API_UNSPECIFIED)
```

Comment:
Seems to be an inconsistent use of brackets here + style guide says to always use brackets for one-liners.